aspnetcore/performance/response-compression/sample/Startup.cs

```diff
@@ -0,0 +1,103 @@
+using Microsoft.AspNetCore.ResponseCompression;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace ResponseCompressionSample
+{
+    public class Startup
+    {
+        #region snippet2
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddResponseCompression(options =>
+            {
+                options.Providers.Add<GzipCompressionProvider>();
+                options.Providers.Add<CustomCompressionProvider>();
+                options.MimeTypes = ResponseCompressionDefaults.MimeTypes.Concat(new[] { "image/svg+xml" });
+            });
+            services.Configure<GzipCompressionProviderOptions>(options => 
+            {
+                options.Level = CompressionLevel.Fastest;
+            });
+        }
+        #endregion
+
+        #region snippet3
+        public void Configure(IApplicationBuilder app)
+        {
+            app.UseResponseCompression();
+
+            app.Map("/testfile1kb.txt", fileApp =>
+            {
+                fileApp.Run(context =>
+                {
+                    ManageVaryHeader(ref context);
+                    context.Response.ContentType = "text/plain";
+                    return context.Response.SendFileAsync("testfile1kb.txt");
+                });
+            });
+
+            app.Map("/trickle", trickleApp =>
+            {
+                trickleApp.Run(async context =>
+                {
+                    ManageVaryHeader(ref context);
+                    context.Response.ContentType = "text/plain";
+                    // Disables compression on net451 because that GZipStream does not implement Flush
+                    context.Features.Get<IHttpBufferingFeature>()?.DisableResponseBuffering();
+
+                    for (int i = 0; i < 20; i++)
+                    {
+                        await context.Response.WriteAsync("a");
+                        await context.Response.Body.FlushAsync();
+                        await Task.Delay(TimeSpan.FromSeconds(1));
+                    }
+                });
+            });
+
+            app.Map("/banner.svg", fileApp =>
+            {
+                fileApp.Run(context =>
+                {
+                    ManageVaryHeader(ref context);
+                    context.Response.ContentType = "image/svg+xml";
+                    return context.Response.SendFileAsync("banner.svg");
+                });
+            });
+
+            app.Run(async context =>
+            {
+                ManageVaryHeader(ref context);
+                context.Response.ContentType = "text/plain";
+                await context.Response.WriteAsync(LoremIpsum.Text);
+            });
+        }
+        #endregion
+
+        #region snippet1
+        private void ManageVaryHeader(ref HttpContext context)
+        {
+            // If the Accept-Encoding header is present, always add the Vary header

```

Comment:
This comment isn't quite accurate. The 2.0 feature operates a little differently than described here. I understand this is probably sufficient for the sample but we should mention the subtleties.